app/templates/newmaps.html

```diff
@@ -8,62 +9,156 @@
+                          $('.linkInsert').attr('href',completeLink);
+                        }
+                      }); 
+                    }
+                    map.addMarkers(markers);     //adds all markers to map  
+                    map.fitZoom();              //and fits map window to show all of them
+                  }
+                  }); }) ();
+              }
+           });
       $('#geocoding_form').submit(function(e){
-
+        map.removeMarkers();
         // Check that zip code is properly formed.
         var regex = /^(\d{5})?$/; // 5-digit number or an empty string.
         var zipCode = $("#address").val().trim(); 
+        console.log(zipCode);
         e.preventDefault();
         if (regex.test(zipCode)) {
             $("#map-search-error").html("");
-        } else {
+        } else { //condition for when an address (not a zipcode) is inserted
+              var new_address = zipCode;
+              map.removeMarkers();
+                  GMaps.geocode({
+                  address: new_address,
+                  callback: function(results, status) {
+                    if (status == 'OK') //add a marker with the address given to mark the user's location
+                    {
+                      var newMarker = results[0].geometry.location; 
+                      myLat = newMarker.lat();
+                      myLng = newMarker.lng();
+                      markers.push({
+                        lat : newMarker.lat(),
+                        lng : newMarker.lng(),
+                        title: "Current Address"
+                      }); 
+                    }
+                    map.addMarkers(markers);  
+                    console.log(markers);    
+                    map.fitZoom();  
+                  }
+                  });
           // Provide helpful error message to user.
           // Clear any errors.
+
+          if (Number.isInteger(zipCode)) {
           $("#map-search-error")
             .html("<small class=error>Zip code must consist of exactly 5 digits (0-9).</small>");
-            return;
+            return; }
+          
+
         }
         // Store search result in database
         var json_data = {
           zipCode: zipCode
         };
-        $.post(url = $SCRIPT_ROOT + '/_search_query', data = json_data);
-
-
+        //cases for different checkboxes
+        var queryString = 'search_query'
+        if ($("#map-filter-by-seniors").is(':checked')) { queryString += '_seniors'; }
+        if ($("#map-filter-by-seniors").is(':checked')) { queryString += '_family';}
+        {
+          queryString += "_seniors";
+        }
+        if ($("#map-filter-by-seniors").is(':checked'))
+        {
+          queryString += "_seniors";
+        }
+        if ($("#map-filter-by-seniors").is(':checked'))
+        {
+          queryString += "_seniors";
```

Comment:
Is this checkbox code currently functional? If not, can you comment it out and explain what it's current behavior is.